torch/quantization/observer.py

```diff
@@ -1072,6 +1073,91 @@ def forward(self, x):
     def calculate_qparams(self):
         raise Exception("calculate_qparams should not be called for NoopObserver")
 
+def _is_observer_script_module(mod, obs_type_name):
+    ''' Returns true if given mod is an instance of Observer script module.
+    '''
+    if isinstance(mod, torch.jit.RecursiveScriptModule):
+        # qualified name looks like '__torch__.torch.quantization.observer.___torch_mangle_2.MinMaxObserver'
+        suffix = mod._c.qualified_name.split('.', 1)[1]
+        name = re.sub(r'\.___torch_mangle_\d+', '', suffix)
+        return obs_type_name in name
+    return False
+
+def _is_activation_post_process(module):
+    return (isinstance(module, torch.quantization.ObserverBase) or
+            isinstance(module, torch.quantization.FakeQuantize) or
+            _is_observer_script_module(module, 'torch.quantization.observer'))
+
+def _is_per_channel_obs_instance(module):
+    if isinstance(module, torch.jit.RecursiveScriptModule):
+        return _is_observer_script_module(module, "torch.quantization.observer.PerChannelMinMaxObserver") or\
+            _is_observer_script_module(module, "torch.quantization.observer.MovingAveragePerChannelMinMaxObserver")
+    else:
+        return isinstance(module, torch.quantization.PerChannelMinMaxObserver) or\
+            isinstance(module, torch.quantization.MovingAveragePerChannelMinMaxObserver)
+
+def _is_per_tensor_obs_instance(module):
+    if isinstance(module, torch.jit.RecursiveScriptModule):
+        return _is_observer_script_module(module, "torch.quantization.observer.MinMaxObserver")
+    else:
+        return isinstance(module, torch.quantization.MinMaxObserver)
+
+def _is_hist_obs_instance(module):
+    if isinstance(module, torch.jit.RecursiveScriptModule):
+        return _is_observer_script_module(module, "torch.quantization.observer.HistogramObserver")
+    else:
+        return isinstance(module, torch.quantization.HistogramObserver)
+
+def _get_attr_names_for_obs(module, name):
+    r"""
+    Return a dict that contains the module attr to fully qualified name
+    in the model. The name should correspond to the attr name in the model
+    state_dict.
+    """
+    qual_names = {}
```

Comment:
As a better engineering task, should we always a have a generic "get_stats" method for observerbase that returns the stats? This way, if we create a new observer, we wouldnt have to change this code.

Reply:
or, we can reuse the state_dict which already does this for us :)